### PR TITLE
amp-layout: move it to builtin + bug fix + example file + visual tests

### PR DESCRIPTION
**Pull request by aghassemi**

### CODE REVIEW

File: validator/validator-main.protoascii [hunk 1]
```diff
@@ -3523,6 +3523,21 @@ tags: {  # <amp-img>
     supported_layouts: RESPONSIVE
   }
 }
+tags: {  # <amp-layout>
+  html_format: AMP
+  html_format: AMP4ADS
+  tag_name: "AMP-LAYOUT"
+  attr_lists: "extended-amp-global"
+  spec_url: "https://www.ampproject.org/docs/reference/components/amp-layout"
+  amp_layout: {
```
Comment (aghassemi) [inline validator/validator-main.protoascii]: @honeybadgerdontcare is there is way to ban immediate children from being Text nodes?

Comment (aghassemi) [reply validator/validator-main.protoascii]: never mind, change the approach so text nodes work as well.

**Comment by honeybadgerdontcare** [comment]
Since this is being added to amp4ads, need to add it to validator_test.cc exception and the amp4ads format document.

**Comment by honeybadgerdontcare** [comment]
Nevermind, I see that you did add it. I'll remove my comment.

**Comment by dvoytenko** [comment]
LGTM. Thanks a lot!

**Pull request merged**